components/layout_2020/flow/inline.rs

```diff
@@ -179,7 +179,10 @@ impl LineUnderConstruction {
                 break;
             }
         }
-        self.justification_opportunities -= spaces_trimmed;
+        self.justification_opportunities = self
+            .justification_opportunities
+            .checked_sub(spaces_trimmed)
+            .unwrap_or(0);
```

Comment:
I think this might be papering over the issue. We shouldn't be subtracting more justification opportunities than we counted up to this point.